### PR TITLE
add promos to Settings and post types

### LINKED ISSUE #48: Add promo section to admin screen
At this point, only the single settings page would make sense. So I'm going to add a sidebar area (most likely) there.

### PR DESCRIPTION
**Pull request by SeanChDavis**
Resolves #48

The goal here was to promote BFCM2019 in various parts of the plugin. In doing so, I needed to do some refactoring that stepped a little outside of the BFCM footprint.

Here's what I've done:

1. Created an admin stylesheet. It only loads on our Settings page. Right now, it only includes the CSS used for the promo.

2. Moved the current `css` and `js` directories into a new `assets` folder. The main reason why I did this is because I needed to add some SVGs for the promo, and those would live inside of an `images` directory. So at this point, it just makes sense to put all of the assets in their own directory. I adjusted resource paths where I found them but please confirm I didn't miss any.

3. Added a function `rc_maybe_display_promotion()` that's used in two places in order to determine whether or not we should be showing a promo. Right now, since this is really just for BFCM, it's based on a start and end time (UTC) that matches when our sale starts and ends. So to test all I've done, got to _Line 129_ in _includes/misc-functions.php_ and...

change `$start = strtotime( '2019-11-29 06:00:00' );`
to `$start = strtotime( '2019-11-04 06:00:00' );`

With the start time changed to a time before now, you'll see the two promos. They are...

4. A sidebar promo to the Settings page. Since the purpose of it conflicts with the existing upsell, I chose to remove the existing one only when the new one is showing. As soon as the sale is over, everything goes back to normal. I refactored much of the structure of this settings page as well as some of the code in the file. This was simply to make the page more flexible for its various uses.

5. An alternate promo to the bottom of the RC metabox on various post types. Again, we already have a promo there. So I used the aforementioned function to determine as a condition for swapping it out for BFCM, and then putting it back once the period is over.

6. Added `RC_PLUGIN_URL` definition just to use while loading images. It's just useful.

### CODE REVIEW

File: includes/settings.php [hunk 1]
```diff
@@ -16,111 +16,199 @@
...
+			</div>
+
+			<div class="rc-sidebar-coupon-section">
+				<label for="rc-coupon-code"><?php _e( 'Use <strong>COUPON CODE</strong> at checkout:', 'restrict-content' ); ?></label>
+				<input id="rc-coupon-code" type="text" value="BFCM2019" readonly>
+				<p class="rc-coupon-note"><?php _e( 'The sale ends the 6th of December at midnight (UTC-6). Use the same coupon code to save on <a href="https://sandhillsdev.com/projects/" target="_blank">our other WordPress plugins</a>.', 'restrict-content' ); ?></p>
+			</div>
+
+			<div class="rc-sidebar-footer-section">
+				<a class="rc-cta-button" href="https://restrictcontentpro.com/pricing/"><?php _e( 'Upgrade Now!', 'restrict-content' ); ?></a>
```
Comment (ashleyfae) [inline includes/settings.php]: Do we want to add campaign data to this URL? The other URLs have them.

Comment (SeanChDavis) [reply includes/settings.php]: @nosegraze Yep! Marketing is getting those together for me and I'll add them to all projects. I keep forgetting that I'm not the one who merges these PRs... so I should make sure y'all know that beforehand lol.

We can let this sit. Once I get the UTM goodies from marketing, I'll be adding it and then I'll re-request your review.

Comment (ashleyfae) [reply includes/settings.php]: @SDavisMedia  Sounds good! Other than that I'm happy with everything.

**Comment by SeanChDavis** [comment]
@nosegraze This is officially done unless you find any issues. I did a little bit of refactoring again just for consistency. But still, nothing is changing about how the plugin behaves outside of BFCM.

I re-requested your review. If all is well, you are free to take everything from here all the way to the next release. Thanks for the help!

**Comment by ashleyfae** [comment]
@SDavisMedia Just made a couple teeny tiny changes: https://github.com/restrictcontentpro/restrict-content/pull/49/commits/a55a9796020b41ed1852e5a95f12b9dd02637e78 All good now!

**Review by ashleyfae — APPROVED** [review]

**Pull request merged**